torch/_export/utils.py

```diff
@@ -42,6 +41,50 @@
 }
 
 
+def _get_shape_env(gm):
+    vals = [
+        node.meta["val"]
+        for node in gm.graph.nodes
+        if node.meta.get("val", None) is not None
+    ]
+
+    fake_mode = _detect_fake_mode_from_gm(gm)
+    if fake_mode is not None:
+        return fake_mode.shape_env
+    for v in vals:
```

Comment:
Should we also assert that these shape_envs matches? Just to add more invariants to our system.

Reply:
Yeah good idea!